### PR TITLE
#218 - Ignore invalid packages on update

### LINKED ISSUE #218: RpmTest.java:43-46: Don't change metadata when invalid...
The puzzle `190-d298c6d5` from #190 has to be resolved: 

https://github.com/artipie/rpm-adapter/blob/19556b01d1294a33853eb6a65b7711bc2539349e/src/test/java/com/artipie/rpm/RpmTest.java#L43-L46

The puzzle was created by @olegmoz on 02-Jun-20. 

Estimate: 30 minutes,  role: DEV. 

If you have any technical questions, don't ask me, submit new tickets instead. The task will be \"done\" when the problem is fixed and the text of the puzzle is _removed_ from the source code. Here is more about [PDD](http://www.yegor256.com/2009/03/04/pdd.html) and [about me](http://www.yegor256.com/2017/04/05/pdd-in-action.html).

### PR DESCRIPTION
**Pull request by olegmoz**
Closes #218

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#223](https://github.com/artipie/rpm-adapter/issues/223) is assigned to @victornoel[/z](https://www.0crat.com/u/victornoel), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/124d5459-0388-48f8-a8bb-f482624c47cb); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by g4s8** [comment]
@olegmoz please resolve merge conflicts

**Comment by olegmoz** [comment]
@g4s8 done

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 1]
```diff
@@ -53,16 +49,19 @@
     private static final String REPOMD = "repodata/repomd.xml";
 
     @Test
-    @Disabled
-    void doesntBrakeMetadataWhenInvalidPackageSent(@TempDir final Path tmp)
+    void doesntBrakeMetadataWhenInvalidPackageSent()
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/rpm/RpmTest.java:52]: @olegmoz this test does not seem accurate enough: except repomd rpm has other metadata files, that should not be changed to. So I think we need todo to extend this test.

File: src/test/java/com/artipie/rpm/RpmTest.java [hunk 2]
```diff
@@ -28,22 +28,18 @@
...
 /**
  * Unit tests for {@link Rpm}.
  *
  * @since 0.9
- * @todo #190:30min Don't change metadata when invalid package is sent.
- *  Currently Rpm is recalculating metadata when an invalid package is sent.
- *  It should not. Correct that and enable the test below.
- *  To do that `InvalidPackageException` should be caught when parsing packages.
  */
 final class RpmTest {
```
Comment (olenagerasimova) [inline src/test/java/com/artipie/rpm/RpmTest.java:44]: @olegmoz also we have to make sure that incremental update behaves the same: please, add todo to add corresponding test method and fix.

**Comment by olegmoz** [comment]
@olenagerasimova I think that you are right and the cases you mentioned should be tested. I just do not think that creating this issues is in scope of this puzzle. I'd suggest you create issues regarding lack of tests for `Rpm` class, because you understand this project way better then me and so know better what needs to be done.

**Comment by victornoel** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/rpm-adapter/issues/223#issuecomment-638360989))

@victornoel Job refused in 9 hours - no penalty, see [§6](http://www.zerocracy.com/policy.html#6)

**Comment by 0crat** [comment]
This pull request [#223](https://github.com/artipie/rpm-adapter/issues/223) is assigned to @paulodamaso[/z](https://www.0crat.com/u/paulodamaso), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/80064c0c-2efe-41a3-b40b-a1f829b94ff8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by olenagerasimova** [comment]
@g4s8 could you please take a look and merge?

**Comment by g4s8** [comment]
@0crat refuse

**Comment by 0crat** [comment]
> @0crat refuse ([here](https://github.com/artipie/rpm-adapter/issues/223#issuecomment-638663095))

@g4s8 The user @paulodamaso[/z](https://www.0crat.com/u/paulodamaso) resigned from [#223](https://github.com/artipie/rpm-adapter/issues/223), please stop working. Reason for job resignation: Order was cancelled

**Comment by 0crat** [comment]
Job `gh:artipie/rpm-adapter#223` is not assigned, can't get performer

**Pull request merged**